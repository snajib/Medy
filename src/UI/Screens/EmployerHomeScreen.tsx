import React from "react";
import { StyleSheet, Text, View, ViewStyle, TextStyle } from "react-native";
import {
  createStackNavigator,
  createAppContainer,
  createBottomTabNavigator
} from "react-navigation";
import { LinearGradient } from "expo-linear-gradient";

export class EmployerHomeScreen extends React.Component {
  render() {
    return (
      <View style={styles.containerStyle}>
        <LinearGradient
          colors={["rgba(86,141,230,1)", "transparent"]}
          style={{
            position: "absolute",
            left: 0,
            right: 0,
            top: 0,
            height: 1200
          }}
        />
        <View style={styles.cardStyle}>
          <View style={styles.centerText}>
            <Text>Home</Text>
          </View>
        </View>
      </View>
    );
  }
}

const containerStyle: ViewStyle = {
  flex: 1,
  backgroundColor: "#5F44BA",
  alignItems: "center",
  justifyContent: "center"
};

const centerText: ViewStyle = {
  flex: 1,
  alignItems: "center",
  justifyContent: "center"
};

const cardStyle: ViewStyle = {
  backgroundColor: "#fff",
  borderRadius: 12,
  shadowColor: "#000",
  shadowOffset: { width: 0, height: 3 },
  shadowOpacity: 0.8,
  shadowRadius: 3,
  height: "70%",
  width: "67%"
};

const linearGradient: ViewStyle = {
  flex: 1,
  paddingLeft: 15,
  paddingRight: 15,
  borderRadius: 5
};

const styles = StyleSheet.create({
  containerStyle,
  cardStyle,
  linearGradient,
  centerText
});

/*TODO: Find a way to create a stack for each screen
   Anything that transitions from the original page of the tab to a separate
   page within the original page of a tab requires a stack

   Exs. 
      Super list inside of Findings tab
      View posting inquiries inside of Postings tab
      View employee cards or QR codes from Calender tab
      View other options in the more tab (micro transactions)
*/

