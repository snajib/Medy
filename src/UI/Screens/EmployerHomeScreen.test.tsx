import React from 'react';
import renderer from 'react-test-renderer';
import { EmployerHomeScreen } from './EmployerHomeScreen';
import { View } from 'react-native';

describe('EmployerHomeScreen', () => {
  it('should render views correctly', () => {
    const homeScreen = renderer.create(<EmployerHomeScreen />);

    expect(homeScreen.root.findAllByType(View).length).toBe(8);
  });

  it('should render with snapshot', () => {
    const employerHomeScreen = renderer.create(<EmployerHomeScreen />);
    expect(employerHomeScreen.toJSON()).toMatchSnapshot();
  });
});
