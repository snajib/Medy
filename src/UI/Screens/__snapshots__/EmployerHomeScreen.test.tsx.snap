// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployerHomeScreen should render with snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#5F44BA",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "marginTop": 80,
      }
    }
    contentInset={
      Object {
        "bottom": 80,
        "left": 0,
        "right": 0,
        "top": 0,
      }
    }
    decelerationRate={0}
    horizontal={false}
    showsVerticalScrollIndicator={false}
    snapToAlignment="start"
    snapToInterval={834}
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 12,
            "height": 884,
            "marginBottom": 40,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 3,
            "width": 650,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text>
            Calender
          </Text>
          <Text />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 12,
            "height": 884,
            "marginBottom": 40,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 3,
            "width": 650,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text>
            Statistics
          </Text>
          <Text />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 12,
            "height": 884,
            "marginBottom": 40,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 3,
            "width": 650,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text>
            Performance
          </Text>
          <Text />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
